### PR TITLE
fix: add thread CTA component

### PR DESCRIPTION
**Pull request by DaoSpirit**
#390 
extended api/google-conversion parser to support CTA component.
extended renderer parser to support CTA component.
implemented UI of CTA section.

### CODE REVIEW

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        href={item.link1.url}>{item.link1.text}</a
+      >
+    {/if}
+
+    {#if item.link2}
+      <a
+        class="relative xs:w-full bg-l1 text-t1 text-button border border-solid border-l5 shadow-secondary-button items-center justify-center p-4 xs:py-2 rounded-xl hover:!border-accent1-active focus:!border-accent1-active dark:bg-l3 dark:border-l3 dark:hover:shadow-secondary-button-hover"
+        href={item.link2.url}>{item.link2.text}</a
+      >
+    {/if}
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:34]: please check for existing components

Comment (georgeciubotaru) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:34]: @DaoSpirit https://github.com/holdex/holdex-venture-studio/tree/main/src/lib/components/Button

Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:34]: Actually I tried with that but some annoying mismatch with secondary button when it comes with light mode

Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: @georgeciubotaru  I've implemented it with existing button.
but secondary button will be different from Figma design.
plz check and leave feedback.

**Review by georgeciubotaru — CHANGES_REQUESTED** [review]
see my comment

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 1 above)
Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: sure

**Comment by DaoSpirit** [comment]
I fixed feedback
@georgeciubotaru

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      position: absolute
+      display: block
+      bottom: 0px
+      right: 50px
+      width: 200px
+      height: 1px
+      background: linear-gradient(90deg, rgba(47, 50, 61, 0.00) 0.5%, #2E83E5 25.56%, #2E83E5 50.63%, rgba(46, 131, 229, 0.25) 75.69%, rgba(47, 50, 61, 0.00) 100.75%)
+      background-repeat: no-repeat
+      @media screen and (max-width: 450px)
+        right: 20px
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:78]: you can combine the common styles

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    .cta-buttons 
+      :global(.cta-button)
+        font-size: 14px
+        line-height: 32px
+        font-weight: 700 !important
+        text-align: center
+      :global(:is([data-theme="light"] .border-secondary))
+        border: 1px solid var(--layer-light-05, #CED1DB)
+      :global(:is([data-theme="dark"] .border-secondary))
+        border: none
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:53]: no need for this one, as the buttons are standable components

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    {#if item.link2}
+      <Button
+        class="cta-button text-t1 border-secondary"
+        variant="secondary"
+        size="large"
+        href={item.link2.url}
+      >
+        {item.link2.text}
+      </Button>
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:37]:
```suggestion
      <Button
        class="cta-button text-t1 border-secondary"
        variant="secondary"
        size="large"
        href={item.link2.url}
        label="{item.link2.text}"
      />
```

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    {#if item.description}
+      <p class="text-cta text-t3 dark:text-t2">{item.description}</p>
+    {/if}
+  </div>
+
+  <div class="ml-auto cta-buttons dark:cta-buttons-dark flex flex-row gap-4 xs:flex-col xs:w-full">
+    {#if item.link1}
+      <Button class="cta-button text-t4" variant="primary" size="large" href={item.link1.url}>
+        {item.link1.text}
+      </Button>
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:26]:
```suggestion
      <Button class="cta-button text-t4" variant="primary" size="large" href={item.link1.url} label="{item.link1.text}"/>
```

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    {#if item.title}
+      <p class="text-cta-title">{item.title}</p>
+    {/if}
+
+    {#if item.description}
+      <p class="text-cta text-t3 dark:text-t2">{item.description}</p>
+    {/if}
+  </div>
+
+  <div class="ml-auto cta-buttons dark:cta-buttons-dark flex flex-row gap-4 xs:flex-col xs:w-full">
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: cleanup unused classes

File: tailwind.config.cjs [hunk 7]
```diff
@@ -247,13 +247,20 @@ module.exports = {
           fontWeight: '400',
         },
       ],
-      cta: [
-        '14px',
+      'cta-title': [
```
Comment (georgeciubotaru) [inline tailwind.config.cjs:250]: why did you update these styles?

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+>
+  <div class="flex flex-col xs:w-full">
+    {#if item.title}
+      <p class="text-cta-title">{item.title}</p>
+    {/if}
+
+    {#if item.description}
+      <p class="text-cta text-t3 dark:text-t2">{item.description}</p>
+    {/if}
+  </div>
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:20]: wrong font styles

**Review by georgeciubotaru — CHANGES_REQUESTED** [review]
see my changes

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 6 above)
Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: All styles are used, nothing to omit

**Comment by DaoSpirit** [comment]
updated feedback @georgeciubotaru 
some parts are not clear.
I left comment, plz check and share opinion.

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 3 above)
Comment (georgeciubotaru) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: @DaoSpirit The button styles are out of your PR scope. Please implement only the styles related to that component

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 8 above)
Comment (georgeciubotaru) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: @DaoSpirit Still the font doesn't match the ones from Figma

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 6 above)
Comment (georgeciubotaru) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: see this one

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 3 above)
Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: but removing this component will not matched with figma design.
is it okay?

Comment (georgeciubotaru) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: @DaoSpirit Did you update your branch with the changes from main branch?

**Review by georgeciubotaru — CHANGES_REQUESTED** [review]
see unresolved comments

**Comment by georgeciubotaru** [comment]
@markholdex, @angelicawill  Can you please take the visual review, all good from my side.

**Review by georgeciubotaru — DISMISSED** [review]

**Comment by zolotokrylin** [comment]
@DaoSpirit how many hours of work did you spend before requesting a review and how many hours in total, including fixing rejected PR?

**Comment by DaoSpirit** [comment]
before review: 8 hours
after review: 4 hours
total: 12 hours

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 9]
```diff
@@ -0,0 +1,47 @@
+<script lang="ts">
+  import type { CTAElement } from '$components/BodyParser/blocks';
+  import Button from '$components/Button/index.svelte';
+  /* eslint-disable @typescript-eslint/no-unused-vars */
+
+  export let item: CTAElement;
+</script>
+
+<div
+  class="relative cta flex flex-row xs:flex-col gap-6 w-full border border-solid border-l4 rounded-xl p-4 bg-l1 dark:bg-l2"
```
Comment (angelicawill) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:10]: should add xs:gap-4 to match the design in desktop view

File: src/lib/components/BodyRenderer/Blocks/cta.svelte [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+    {#if item.title}
+      <p class="text-paragraph-l text-t1 font-bold">{item.title}</p>
+    {/if}
+
+    {#if item.description}
+      <p class="text-paragraph-l text-t3">{item.description}</p>
+    {/if}
+  </div>
+
+  <div class="ml-auto flex flex-row gap-4 xs:flex-col xs:w-full py-1.75">
```
Comment (angelicawill) [inline src/lib/components/BodyRenderer/Blocks/cta.svelte:22]:
```suggestion
  <div class="ml-auto flex flex-row gap-2 xs:gap-4 xs:flex-col xs:w-full">
```

**Review by angelicawill — COMMENTED** [review]
@DaoSpirit There's some minor pixel perfect issue, please check my comments. Thanks!

**Review by zolotokrylin — CHANGES_REQUESTED** [review]
@DaoSpirit see Angelica's comments above.

@angelicawill in the future use REJECT feature in that case. Thanks

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 10 above)
Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: suggested change doesn't match with design, I think

Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: @DaoSpirit may I know which part?

Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: padding top&bottom doesn't work @angelicawill

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 9 above)
Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:10]: it seems gap-6 for desktop view @angelicawill

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 10 above)
Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: @DaoSpirit there's no padding on the button wrapper in the design
CleanShot 2023-12-21 at 14 14 31@2x

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 9 above)
Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:10]: CleanShot 2023-12-21 at 14 15 23@2x

@DaoSpirit the gap is 16px, which is gap-4 I believe

**Comment by markholdex** [comment]
@DaoSpirit can you please update us on the status of this PR? Do you plan to complete it? 

If so, I would appreciate it if you would respect our Guidelines and don't leave a PR handing without updates. Also, the PR completion time should not exceed 4 hours. I propose you wrap things up soon, otherwise, the PR will be closed without merge. Thanks! 🙏

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 10 above)
Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: image
fixed result is screenshot.

Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: as we can see in your screenshot,
gap-4 xs:gap-2 is matched with design
not gap-2 xs:gap-4
@angelicawill

**Comment by zolotokrylin** [comment]
@markholdex there are updates. See comments above your message.

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 10 above)
Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: @DaoSpirit you're right, I got mixed up. gap-4 xs:gap-2 is the right one. sorry for that.

In this case my suggested change will be 
  <div class="ml-auto flex flex-row gap-4 xs:gap-2 xs:flex-col xs:w-full">

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 9 above)
Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:10]: and here my suggested change will be adding `xs:gap-6 gap-4` to accommodate the desktop and mobile view design

**Comment by zolotokrylin** [comment]
@georgeciubotaru @DaoSpirit @angelicawill please "resolve" the resolved comments. Those which are not resolved, resolve ASAP and let's move on to the next step of the interview.

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 10 above)
Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: @angelicawill  we should add py-2 to match the design.
I'll share screenshot. and button size is too big
image

Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: @DaoSpirit I see. I just checked, and seems that it's an issue from the button component, which stretches in height to fit the space. I'll fix the button component to include `height:fit-content`, and you can remove the py-2. I'll create a separate PR for this

Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: okay
so should I push updated one without fixing button height? @angelicawill

Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: @DaoSpirit yes, but please add vertical center aligning so the button component will be positioned in the middle when the height changes. thank you!

Comment (angelicawill) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: @DaoSpirit fyi I opened a PR that fixes the button issue, #555. It hasn't been merged yet, but you can test out the button style to see if the CTA match with the design now. 🙏

Comment (DaoSpirit) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte:22]: I checked it @angelicawill 
it matched with the design.
but adding my-auto to align vertical center shows the same result with changed button style.
I'll push commit and plz check

**Review by angelicawill — APPROVED** [review]
All good from my side.

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 2 above)
Comment (markholdex) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: @DaoSpirit have you implemented these changes?

File: src/lib/components/BodyRenderer/Blocks/cta.svelte (hunk 3 above)
Comment (markholdex) [reply src/lib/components/BodyRenderer/Blocks/cta.svelte]: @DaoSpirit is this comment resolved?

File: src/lib/components/BodyRenderer/index.svelte [hunk 11]
```diff
@@ -1,12 +1,30 @@
...
+          text: 'Button 1',
+          url: 'https://google.com',
+        },
+        link2: {
+          text: 'Button 2',
+          url: 'https://google.com',
+        },
+      },
+    },
+  ];
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/index.svelte:23]:
```suggestion
```

File: src/lib/components/BodyRenderer/index.svelte [hunk 12]
```diff
@@ -1,12 +1,30 @@
...
 </script>
 
-{#each blocks as item, index}
-  <Item item={item} index={index} />
+{#each addtionalBlocks as item, index}
+  <Item {item} {index} />
 {/each}
```
Comment (georgeciubotaru) [inline src/lib/components/BodyRenderer/index.svelte:28]:
```suggestion
{#each blocks as item, index}
  <Item {item} {index} />
{/each}
```

**Review by georgeciubotaru — APPROVED** [review]

**Pull request merged**